pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -365,3 +413,42 @@ func loadRegistryConf(configPath string) (*tomlConfig, error) {
 	err = toml.Unmarshal(configBytes, &config)
 	return config, err
 }
+
+// IsSecureIndex returns true if indexName matches an element of CIDRs. indexName can be of form `host:port` or `host`
+// where the `host` part can be either a domain name or an IP address. If it is a domain name, then it will be
+// resolved to IP addresses for matching. If resolution fails, false is returned.
+func (s *SysRegistries) IsSecureIndex(indexName string) bool {
```

Comment:
We have _just_ modified consumers of `sysregistriesv2` to use `FindRegistry` (note the prefix semantics - it’s not just host[:port] values!) and check `.Insecure` on the result.

It’s fine to change the API, but we can’t have two different ways to ask the same question which return different results, the way right now `IsSecureIndex` and `FindRegistry().Insecure` do.

My initial guess would be to somehow embed this into the `FindRegistry` prefix matching, with specific repo/namespace configs taking precedence, as they currently do, and then adding the CIDR matching roughly at the level of host:port matches (I guess, as a last fallback after that) — but that might well be the wrong way to do it.

Reply:
Doing this in `FindRegistry()` might be a tad bit complicated. If we want to be able to use `FindRegistry().Insecure` instead of `IsSecureIndex`, then we would have to convert the CIDR notation and host:port notation to type Registry. One was I was thinking was to add a field `isCIDR` and if that is true then you have a field `IPNet`, which would be populated. Just a thought.
I am on PTO from today onwards and will have limited access to internet as I am travelling. I can try to get back to this but since this is high priority could @mtrmac or @vrothberg please take over.

Reply:
> Doing this in `FindRegistry()` might be a tad bit complicated. If we want to be able to use `FindRegistry().Insecure` instead of `IsSecureIndex`, then we would have to convert the CIDR notation and host:port notation to type Registry.

This is a great point; yes, `FindRegistry` would have to manufacture a `Registry` value for the host that matched the CIDR range (rather than return the whole range). That’s not _obviously_ the right thing to do, but it also seems not obviously unreasonable to me.

Alternatively we could instead design the API to “answer specific questions”, the way `IsSecureIndex` works (add `GetMirrors`/`IsBlocked`…), and probably eliminate `FindRegistry` and hide the `Registry` type altogether.

I slightly prefer the `FindRegistry` approach because it clearly centralizes the lookup semantics, and with plausible client implementations, the O(N) lookup of the matching configuration entry can happen only once instead on each `IsSecure`/`IsBlocked`/`GetMirrors`/… call.

But it’s not obviously the right answer, hoping for others to weigh in.

> One was I was thinking was to add a field `isCIDR` and if that is true then you have a field `IPNet`, which would be populated. Just a thought.

Do the callers of `FindRegistry` actually need to know whether the data matched via a CIDR at all? e.g. the current / soon-to-be-introduced callers just consult `.Insecure` and don’t use the `URL` entry at all; and `FindRegistry` could, when matching a CIDR, generate a `Registry` value that contains a value based on the specific IP address (not even the range) in `URL`/`Prefix`.